### ISSUE TITLE
Cordova 3.5.1 on steroids (appgyver.com)

### ISSUE THREAD

**Issue by sotoik** [willemmulder/Diont-for-Cordova#7]
I have a nodejs diont running and other diont clients on nodejs are discovering it just fine. When I use diont for cordova on the steroids platform the "serviceAnnounced" event never fires. I use for both the nodejs and for the cordova your usage examples.

**Comment by willemmulder** [comment]
Does it work the other way around? So announcing a service on mobile, and then checking on the nodejs server if the "serviceAnnounced" event fires?

**Comment by sotoik** [comment]
No. On the developer console I see the diont object just fine. It just is empty. If I annouce a service on the mobile then it shows only on the mobile, not on any other device that has a diont server running. Some times the event fires (only for the service on the mobile) but the serviceInfo object is empty most of the time.

**Comment by willemmulder** [comment]
Hmm, sounds like issues with the router that doesn't pass the UDP packets
along properly.

What happens if you announce the services on nodejs via broadcast instead
of the normal multicast?
See https://github.com/willemmulder/Diont#broadcast



2015-10-12 9:56 GMT+02:00 sotoik <notifications@github.com>:

> No. On the developer console I see the diont object just fine. It just is
> empty. If I annouce a service on the mobile then it shows only on the
> mobile, not on any other device that has a diont server running. Some times
> the event fires (only for the service on the mobile) but the serviceInfo
> object is empty most of the time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/willemmulder/Diont-for-Cordova/issues/7#issuecomment-147320797>
> .
>



-- 
http://willemmulder.net

**Comment by sotoik** [comment]
Between nodejs servers and clients on the same WIFI network everything is discovered. The only problem is with the android device that does not get discovered and does not discover any services.

**Comment by willemmulder** [comment]
The android device needs to be on the same WIFI network as well. It is,
right?

If so, then as I said, could you try using the broadcast option?
It might help to get the UDP messages across the WIFI router to the Android
device so that Diont for Cordova picks it up.

2015-10-12 11:33 GMT+02:00 sotoik <notifications@github.com>:

> Between nodejs servers and clients on the same WIFI network everything is
> discovered. The only problem is with the android device that does not get
> discovered and does not discover any services.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/willemmulder/Diont-for-Cordova/issues/7#issuecomment-147347917>
> .
>



-- 
http://willemmulder.net

**Comment by sotoik** [comment]
I have tried the broadcast option (forgot to mention) and it's the same. The router moves UDP packets between the nodejs machines on the same wifi but not on the Android device, also on the same wifi.

**Comment by willemmulder** [comment]
Then I'm afraid I'm a bit out of ideas... Ideally, we want to check if the UDP messages arrive on the Android phone. With something like https://play.google.com/store/apps/details?id=com.googlecode.networklog or similar maybe..?

**Comment by sotoik** [comment]
I will try it out and Inform you.

**Comment by sotoik** [comment]
Some time I get "Diont is not defined" when I inspect the app from Chrome. This is not always the case thought.

**Comment by willemmulder** [comment]
Hi @sotoik that might be because you don't run within the Cordova onDeviceReady event?

It should be like this:

``` javascript
// Cordova standard - run once Cordova and the device is ready
document.addEventListener("deviceready", onDeviceReady, false);

function onDeviceReady() {
    var diont = Diont();
}
```

I also added this to the Readme now. Does this solve your issue?

**Comment by ghost** [comment]
I will test it and report back to you.

2015-10-19 12:25 GMT+03:00 Willem Mulder notifications@github.com:
